### DISCUSSION TITLE
[RSC - client directive] : Not working has expected

### DISCUSSION THREAD

**Discussion by StephaneIzi** [contentlayerdev/contentlayer#423] [Q&A]
Hi,  first thank you for this amazing repository  ❤️ . 

`contentlayer` has been really useful to me lately. 

TLDR : Should i expect `contentlayer` to handle React Component with `client directive` inside MDX ? 

-----------------

I want to use `contentlayer` to hold all my content (`mdx` files)

The stack is : 

- ` nextJS 13`
- `contentlayer`
- `tailwind`

Here is a repo with the stack : https://github.com/Stephane-Pires/contentlayer-mdx-client-directive

I started with MDX using @next/mdx (commit : 1ef8551c6206eb44f0f1717f2f8c32eef7da22d0) , everything was working fine, i was able to import Component with and without `use client` directive directly inside my `mdx` files.  (Below a `gif` of the working `<Counter />` Button) 

file : https://github.com/Stephane-Pires/contentlayer-mdx-client-directive/blob/main/content/hello.mdx

https://user-images.githubusercontent.com/124583979/232328285-44aed85b-9bca-46b8-8634-d82bd367a085.mov

Then i tryied MDX using @contentlayer & @next-contentlayer(commit : a3c027e455de3fda8b329ad2e2113b0c38c042e4) 

- I configured `contentlayer`

- Removed the `<Counter />` component from the MDX

- generate the `.contentlayer` with the CLI `npx contentlayer build` due to #415 (i think)

<img width="525" alt="Screenshot 2023-04-16 at 19 08 00" src="https://user-images.githubusercontent.com/124583979/232328771-eb34c6d8-1718-40e3-8d0a-4cfb97e288f9.png">

Everything is working has expected. 

Things started to go off track when i added the `<Counter />` component inside `MDX` files. 

I tryied : 

- commit : 4761f0852b87a30de8de554e5155d3b3cc00015b
- commit : 353b94051bc2b1b5d6d55d6e290ad3c389776ed9

From my understanding so far, `contentlayer` could handle Client Component 

https://github.com/contentlayerdev/contentlayer/releases

> Note: While it's theoretically also possible to use Contentlayer combined with the 'use client' approach, we don't recommend it as it massively increases page sizes and thus the page load time.

My question is :  Should i expect `contentlayer` to handle React Component with `client directive` inside MDX ? 

P.S : I really want to use `contentlayer` the `types` generation is a big plus for me ! 

Thanks for reading !

**Comment by schickling** [comment]
Possibly related to this https://github.com/contentlayerdev/contentlayer/issues/309#issuecomment-1485490054
